### PR TITLE
feat: vyper 0.4.0 support

### LINKED ISSUE #33: Decorators do not appear commented out
Comments start with `#` and usually the whole line turns comment-colored (gray).
However, decorators, such as visibility-decorators, do not appear commented out.

I think it is best to show a picture here.

Screenshot 2024-01-26 at 09 37 52

Notice the decorator remains red even though it is prefixed with `#` and _is_ commented out from a technical perspective but the visual keeps throwing me off - I have repeatedly been like "oops forgot to comment out the decorator!" It has happened enough times I felt compelled to open this issue.

### PR DESCRIPTION
**Pull request by trocher**
As detailed in the CHANGELOG, bring the following changes:

- new: logo for Vyper files in the explorer view
- fix: compilation for Vyper 0.4.x
- fix: don't compile interface files
- fix: correct static built-in hover information to match Vyper
- fix: prevent function decorator from being highlighted when used in a comment (fixes #33)
- updated snippets to Vyper 0.3.x and 0.4.x
- updated language spec fetcher to Vyper 0.4.x
- updated syntax highlighting to Vyper 0.4.x
- updated the Vyper logo
- updated the banner theme on the marketplace to match Vyper colors
- removed MythX

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
     "displayName": "Vyper",
     "description": "Ethereum Vyper language support for Visual Studio Code",
     "license": "MIT",
-    "version": "0.0.15",
+    "version": "0.0.16",
```
Comment (pcaversaccio) [inline package.json:6]: how about a `0.1.0` release?

File: src/features/compile.js [hunk 2]
```diff
@@ -68,7 +65,15 @@ function checkVyper(source_file, callback) {
 
 // Execute vyper for single source file
 function execVyper(source_path, callback) {
-    const formats = ["abi", "bytecode", "bytecode_runtime"];
+    let formats;
+    if (compiler.version.startsWith("0.4")) {
+        // In 0.4.x, Vyper modules might be valid module to be imported 
+        // but not valid contracts to compile into bytecode, hence we stop
+        // the compilation at annotated_ast.
```
Comment (pcaversaccio) [inline src/features/compile.js:72]: I guess it doesn't hurt to add this info to the `README`.

File: scripts/fetch_vyper_language_spec.py [hunk 3]
```diff
@@ -5,138 +5,203 @@
...
-    interface_decl = ["implements"]
-    special_vars = ["log","msg","block"] #add tx and chain? /DBIT
-
+    modifiers_safe = [
+        "nonreentrant",
+        "internal",
+        "view",
+        "pure",
+        "private",
+        "constant",
```
Comment (pcaversaccio) [inline scripts/fetch_vyper_language_spec.py:37]: `immutable` missing no?

File: scripts/fetch_vyper_language_spec.py [hunk 4]
```diff
@@ -5,138 +5,203 @@
...
+        "EMPTY_BYTES32",
+        "MAX_INT128",
+        "MIN_INT128",
+        "MAX_DECIMAL",
+        "MIN_DECIMAL",
+        "MIN_UINT256",
+        "MAX_UINT256",
+    ]  # kept for backward compatibility
+    modules = ["implements", "uses", "initializes", "exports"]
+    special_vars = ["log", "msg", "block", "tx", "chain", "extcall", "staticcall"]
```
Comment (pcaversaccio) [inline scripts/fetch_vyper_language_spec.py:68]: what about the `__interface__` dunder? where should we put that one?

File: scripts/fetch_vyper_language_spec.py (hunk 3 above)
Comment (trocher) [reply scripts/fetch_vyper_language_spec.py:37]: yes, still handled by `vyper.ast.identifiers.RESERVED_KEYWORDS` but adding for completness.

File: src/features/compile.js (hunk 2 above)
Comment (trocher) [reply src/features/compile.js:72]: updated in the Q&A section

**Pull request merged**